### PR TITLE
Add a timeout parameter and report timeouts

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #249

### CODE REVIEW

File: pulpcore/cli/common/config.py [hunk 1]
```diff
@@ -33,6 +43,13 @@
         default=False,
         help=_("Trace commands without performing any unsafe HTTP calls"),
     ),
+    click.option(
+        "--timeout",
+        "-T",
+        type=int,
+        default=0,
+        help=_("Time to wait for backgroud tasks, set to 0 to wait infinitely"),
```
Comment (ggainey) [inline pulpcore/cli/common/config.py:51]: background

**Review by ggainey — CHANGES_REQUESTED** [review]

**Comment by ggainey** [comment]
Post-mypy-pin PR, this still fails lint, and I'm not sure why :(

**Comment by mdellweg** [comment]
> Post-mypy-pin PR, this still fails lint, and I'm not sure why :(

I believe gh runs those against the branch tips and not trial merges, so it needs rebasing.

**Comment by mdellweg** [comment]
@ggainey ping

**Review by ggainey — APPROVED** [review]

**Pull request merged**